### PR TITLE
SWC-6498

### PR DESCRIPTION
**Pull request by jay-hodgson**
Define Content-Security-Policy

### CODE REVIEW

File: src/main/webapp/Portal.html [hunk 1]
```diff
@@ -504,9 +509,8 @@
       head.load(cdnEndpoint + 'generated/prop-types.min.js')
       head.load(cdnEndpoint + 'generated/react-measure.js')
       head.load(cdnEndpoint + 'generated/plotly.min.js')
-      head.load(
-        'https://unpkg.com/react-plotly.js@2.2.0/dist/create-plotly-component.min.js',
-      )
+      head.load(cdnEndpoint + 'js/2.2.0_create-plotly-component.min.js')
+      head.load(cdnEndpoint + 'generated/react-jsonschema-form.js')
```
Comment (nickgros) [inline src/main/webapp/Portal.html]: react-jsonschema-form should stay removed from the list of scripts to load, no?

Can we add react-plotly.js to the package.json and add build actions to copy the react-plotly.js dependency like we do for the other scripts (so it is captured when Dependabot scans our package.json/yarn.lock files?)

File: src/main/webapp/Portal.html [hunk 2]
```diff
@@ -8,6 +8,11 @@
     />
     <meta name="author" content="Sage Bionetworks, info@sagebase.org" />
     <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+
+    <meta
+      http-equiv="Content-Security-Policy"
+      content="default-src 'self' data: 'unsafe-inline' 'unsafe-eval' *.synapse.org *.sagebase.org *.googleapis.com *.gstatic.com *.googletagmanager.com cdn.statuspage.io s3.amazonaws.com www.google-analytics.com kh896k90gyvg.statuspage.io 127.0.0.1:9876"
```
Comment (nickgros) [inline src/main/webapp/Portal.html]: This seems like an appropriate list. Just wondering how you came up with it?

Comment (jay-hodgson) [reply src/main/webapp/Portal.html]: Reviewing this file, and trial and error (spinning up the client and looking for console messages indicating that the browser blocked a resource due to the csp)

File: src/main/webapp/Portal.html (hunk 1 above)
Comment (jay-hodgson) [reply src/main/webapp/Portal.html]: react-jsonschema-form should certainly stay removed, this must have been a bad complex merge.  I'll delete it

Comment (jay-hodgson) [reply src/main/webapp/Portal.html]: And you're absolutely right regarding create-plotly-component.min.js from react-plotly.  This is not a special case

File: src/main/webapp/Portal.html (hunk 2 above)
Comment (nickgros) [reply src/main/webapp/Portal.html]: Is it possible to embed external images in Wiki markdown? Wonder if we need to add `img-src *`. [MDN reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src).

Comment (nickgros) [reply src/main/webapp/Portal.html]: Would this also affect the contents of an `iframe`d HTML preview? I am not currently sure what functionality remains for that feature

Comment (jay-hodgson) [reply src/main/webapp/Portal.html]: Nice catch with the img-src, I'll add it

**Review by nickgros — APPROVED** [review]

**Pull request merged**